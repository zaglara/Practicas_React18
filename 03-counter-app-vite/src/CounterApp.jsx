import PropTypes from 'prop-types';
import { useState } from 'react';

export const CounterApp = ({value}) => {

    const [counter, setCounter] = useState(value);

    const handleAdd = () => {
        setCounter(counter + 1);
    }

    const handleSubstract = () => {
        setCounter(counter - 1);
    }

    const handleReset= () => {
        setCounter(value);
    }

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <button onClick={handleAdd}>+1</button>
      <button aria-label='btn-reset' onClick={handleReset}>Reset</button>
      <button onClick={handleSubstract}>-1</button>
    </>
  )
}

CounterApp.propTypes = {
    value: PropTypes.number
}