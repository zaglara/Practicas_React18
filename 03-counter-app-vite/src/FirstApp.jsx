import PropTypes from 'prop-types';

export const FirstApp = ({title, subtitle, name}) => {

    return (
        <>
            <h1 data-testid="test-title">{title}</h1>
            <p>{subtitle}</p>
            <p>{subtitle}</p>
            <p>{name}</p>
        </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    title: 'No hay titulo',
    subtitle: 'No hay subtitulo',
    name: 'Leiza'
}