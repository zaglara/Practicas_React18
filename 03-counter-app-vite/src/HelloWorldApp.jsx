const HelloWorldApp = () => {
    return (<h1>Hola Mundo</h1>);
}
export default HelloWorldApp;