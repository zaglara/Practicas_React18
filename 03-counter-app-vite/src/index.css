html, body {
    background-color: #21232a;
    color: white;
    font-family: Helvetica, Arial, sans-serif;
    font-size: 1.3rem;
    padding: 20px;
}

button {
    font-size: 20px;
    height: 3rem;
    padding: 10px;
    margin-right: 1rem;
    width: 5rem;
}