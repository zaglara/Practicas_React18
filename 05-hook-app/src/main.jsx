import React from 'react';
import ReactDOM from 'react-dom/client';
import { BrowserRouter } from "react-router-dom";

//import { HooksApp } from './HooksApp';
//import { CounterApp } from './01-useState/CounterApp';
//import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook';
//import { FormWithCustomHook } from './02-useEffect/FormWithCustomHook';
//import { SimpleForm } from './02-useEffect/SimpleForm';
//import { MultipleCustomHooks } from './03-examples/MultipleCustomHooks';
//import { FocusScreen } from './04-useRef/FocusScreen';
//import { Layout } from './05-useLayoutEffect/Layout';
//import { CallbackHook } from './06-memos/CallbackHook';
//import { MemoHook } from './06-memos/MemoHook';
//import { Memorize } from './06-memos/Memorize';
//import { Padre } from './07-tarea-memo/Padre';
//import './08-useReducer/intro-reducer';
//import { TodoApp } from './08-useReducer/TodoApp';
import { MainApp } from './09-useContext/MainApp';
import './index.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  //<React.StrictMode>
    <BrowserRouter>
      <MainApp />
    </BrowserRouter>
  //</React.StrictMode>
)
