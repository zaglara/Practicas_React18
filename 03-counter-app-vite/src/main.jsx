import React from 'react';
import ReactDOM from 'react-dom/client';
//import { CounterApp } from './CounterApp';
import { FirstApp } from './FirstApp';
//import HelloWorldApp from './HelloWorldApp';

import './index.css';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {//<CounterApp value={3} /
        }
        <FirstApp title='Hola, Soy Leiza' />
    </React.StrictMode>
)