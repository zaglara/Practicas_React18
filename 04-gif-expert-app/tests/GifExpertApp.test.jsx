import {fireEvent, render, screen, waitFor} from '@testing-library/react';
import { GifExpertApp } from '../src/GifExpertApp';

describe('Pruebas de GifExpertApp', () => {

    test('Debe hacer match con el snapshot', () => {
        const { container } = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
    });

    test('Se muestra la primera categoria default', () => {
        render(<GifExpertApp />);
        expect(screen.getByText('One Punch'));
    });

    test('Debe agregarse una categoría nueva', () => {
        const inputValue = 'Goku';

        render(<GifExpertApp />);

        fireEvent.input(screen.getByRole('textbox'), {target: {value: inputValue}});
        fireEvent.submit(screen.getByRole('form'));

        expect(screen.getByText(inputValue));
    });

    test('No debe agregarse una categoría repetida', () => {
        const inputValue = 'One Punch';

        render(<GifExpertApp />);

        fireEvent.input(screen.getByRole('textbox'), {target: {value: inputValue}});
        fireEvent.submit(screen.getByRole('form'));

        expect(screen.getAllByText(inputValue).length).toBe(1);
    });

    test('Se muestran los gifs default', async() => {
        render(<GifExpertApp />);
        await waitFor(
            () => expect(screen.getAllByRole('img').length).toBeGreaterThan(0)
        );
        
        expect(screen.getAllByRole('img').length).toBeGreaterThan(0);
    });

    test('Se muestra una nueva categoria con sus gifs', async() => {
        const inputValue = 'Goku';

        render(<GifExpertApp />);

        fireEvent.input(screen.getByRole('textbox'), {target: {value: inputValue}});
        fireEvent.submit(screen.getByRole('form'));

        await waitFor(
            () => expect(screen.queryAllByText('Cargando...').length).toBe(0)
        );

        expect(screen.getAllByRole('heading', {level:3}).length).toBe(2);
        expect(screen.queryAllByText('Cargando...').length).toBe(0);

    });

});