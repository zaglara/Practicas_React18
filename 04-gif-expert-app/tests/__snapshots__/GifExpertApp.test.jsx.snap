// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifExpertApp Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
